### DISCUSSION TITLE
Is there any layer or model could update the edge but only the node and learn an embedding of edge and node?

### DISCUSSION THREAD

**Discussion by StefanIsSmart** [pyg-team/pytorch_geometric#8742] [Q&A]
As the title is.
Most of the models only focus on how to learn an embedding of nodes and update the node features.
Are there any layers or models that could update the edge and learn the embedding of edges in PyG?

**Comment by rusty1s** [comment]
You can indeed also do this with the `MessagePassing` interface, see the `edge_update` function. In this case, your GNN layer would return both node and edge features, that get consumed in follow-up layers. There doesn't exist a layer though which does that yet. The only thing that comes close is the `MetaLayer`.
